### PR TITLE
add basedownloadcontroller unit tests

### PR DESCRIPTION
**Pull request by kdai7**

### CODE REVIEW

File: lib/MockDownloadController.js [hunk 1]
```diff
@@ -0,0 +1,32 @@
+const MockKubeResourceMeta = require('../lib/MockKubeResourceMeta');
+const BaseDownloadController = require('./BaseDownloadController');
+const log = require('./bunyan-api').createLogger();
+const fs = require('fs-extra');
+
+module.exports = class MockDownloadController extends BaseDownloadController {
+  constructor(eventData, kubeData) {
+    let params = {};
```
Comment (adamkingit) [inline lib/MockDownloadController.js:8]:
```suggestion
    const params = {};
```

File: lib/MockKubeResourceMeta.js [hunk 2]
```diff
@@ -6,25 +9,50 @@ module.exports = class MockKubeResourceMeta {
...
-      kind: reqOpt.uri.kind,
-      name: reqOpt.uri.name,
-      namespace: reqOpt.uri.namespace,
-      labelSelector: reqOpt?.qs?.labelSelector
-    };
-
-    const res = await this.kubeGetResource(ref);
-    return res;
+    try {
+      let uri = JSON.parse(reqOpt.uri);
```
Comment (adamkingit) [inline lib/MockKubeResourceMeta.js:13]:
```suggestion
      const uri = JSON.parse(reqOpt.uri);
```

File: test/basedownloadcontroller-tests.js [hunk 3]
```diff
@@ -0,0 +1,379 @@
...
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+const assert = require('chai').assert;
+const Controller = require('../lib/MockDownloadController');
+const clone = require('clone');
+const objectPath = require('object-path');
+
+let kubeData = {};
```
Comment (adamkingit) [inline test/basedownloadcontroller-tests.js:22]:
```suggestion
const kubeData = {};
```

File: test/basedownloadcontroller-tests.js [hunk 4]
```diff
@@ -0,0 +1,379 @@
...
+
+  it('Update single request option reconcile children', async function () {
+    const controller = setupController(eventData);
+    await controller.execute();
+
+    // update request option
+    let eventDataUpdate = clone(kubeData['RemoteResource'][0]);
+    eventDataUpdate = {
+      object: eventDataUpdate
+    };
```
Comment (adamkingit) [inline test/basedownloadcontroller-tests.js:289]:
```suggestion
    const eventDataUpdate = {
      object: clone(kubeData['RemoteResource'][0])
    };
```

File: test/basedownloadcontroller-tests.js [hunk 5]
```diff
@@ -0,0 +1,379 @@
...
+  });
+
+  it('Update single request option reconcile children false', async function () {
+    const controller = setupController(eventData2);
+    await controller.execute();
+
+    let eventDataUpdate = clone(kubeData['RemoteResource'][0]);
+    eventDataUpdate = {
+      object: eventDataUpdate
+    };
```
Comment (adamkingit) [inline test/basedownloadcontroller-tests.js:323]:
```suggestion
    const eventDataUpdate = {
      object:  clone(kubeData['RemoteResource'][0])
    };
```

**Review by adamkingit — COMMENTED** [review]
I like it overall. Minor suggestions.

File: test/basedownloadcontroller-tests.js (hunk 3 above)
Comment (kdai7) [reply test/basedownloadcontroller-tests.js:22]: gets reset in afterEach, so can't change to const

**Review by adamkingit — APPROVED** [review]

**Pull request merged**